src/main/java/frc/robot/hardware/mechanisms/MechanismSimulation.java

```diff
@@ -2,29 +2,21 @@
 
 import edu.wpi.first.math.geometry.Rotation2d;
 
-public abstract class MechanismSimulation {
+public interface MechanismSimulation {
 
-	private final double gearRatio;
+	double getGearRatio();
 
-	MechanismSimulation(double gearRatio) {
-		this.gearRatio = gearRatio;
-	}
+	Rotation2d getRotorPosition();
 
-	public Rotation2d getRotorPosition() {
-		return getSystemPosition().times(gearRatio);
-	}
+	Rotation2d getRotorVelocityAnglesPerSecond();
 
-	public Rotation2d getRotorVelocityAnglesPerSecond() {
-		return getSystemVelocityAnglesPerSecond().times(gearRatio);
-	}
 
+	Rotation2d getSystemPosition();
 
-	public abstract Rotation2d getSystemPosition();
+	Rotation2d getSystemVelocityAnglesPerSecond();
 
-	public abstract Rotation2d getSystemVelocityAnglesPerSecond();
+	void setInputVoltage(double voltage);
```

Comment:
this just doesn't make sense. If we need to specify that velocity are in Rotation2d/sec, why isn't it VoltageVolt? 

ik this problematic naming occurs outside to this PR. But I wouldn't want to approve inconsistent code.